src/Utils.php

```diff
@@ -90,26 +90,38 @@ public static function debugResource($value = null)
      */
     public static function chooseHandler(): callable
     {
-        $handler = null;
-        if (\function_exists('curl_multi_exec') && \function_exists('curl_exec')) {
-            $handler = Proxy::wrapSync(new CurlMultiHandler(), new CurlHandler());
-        } elseif (\function_exists('curl_exec')) {
-            $handler = new CurlHandler();
-        } elseif (\function_exists('curl_multi_exec')) {
-            $handler = new CurlMultiHandler();
+        $handler = self::getHandler();
+        if (\ini_get('allow_url_fopen')) {
+            return $handler ? Proxy::wrapStreaming($handler, new StreamHandler()) : new StreamHandler();
         }
 
-        if (\ini_get('allow_url_fopen')) {
-            $handler = $handler
-                ? Proxy::wrapStreaming($handler, new StreamHandler())
-                : new StreamHandler();
-        } elseif (!$handler) {
+        if (!$handler) {
             throw new \RuntimeException('GuzzleHttp requires cURL, the allow_url_fopen ini setting, or a custom HTTP handler.');
         }
 
         return $handler;
     }
 
+    /**
+     * @return callable|CurlHandler|CurlMultiHandler|null
```

Comment:
wrong phpdoc. should be `(callable(\Psr\Http\Message\RequestInterface, array): \GuzzleHttp\Promise\PromiseInterface)|null`

Reply:
Changed, it was generated automatically by PHP storm.